### DISCUSSION TITLE
esp32 programin problem ,pls help..

### DISCUSSION THREAD

**Discussion by ddijanic** [forkineye/ESPixelStick#502] [General]
ESPixelStick_Firmware-4.0 beta. 2-3-4

ESP-WROOM-32 and esp-wroom-32u

**Binary release or compiled yourself?**

**win 10 64 bit 1909**

**chroom 99.0.4844.51)**

*NTU-RG-1421G-Wac:rev.A1**

1   
I tried ESPixelStick_Firmware-4.0-beta 2 and 3 and 4, all the same error
https://ibb.co/rf9rHzp

2 flash_download_tool_3.9.2 works but. 
https://prnt.sc/sLTHR7Guwt09

3 visucal stidio code flash works but again HTML error
https://prnt.sc/sLTHR7Guwt09

I did uplod FILESYSTEM Image , whit  visucal stidio code 
https://anotepad.com/notes/5enmgmjm

AND whit arduino esp32 sketch data upload
https://anotepad.com/notes/56agwn8f

after reboot,  the same error  **Page Not found**

serial monitor log off Page Not found

https://prnt.sc/LbbKfmPM18zA

i tried making DATA with gulp and poslatri with arduino nothing.
i tried to build filesystem image with visucal stidio code and nothing.
i tried data from folder java flash tool, and send via arduino, nothing.

I tried beta 2, bate 3, beta 4.
on both ESPs and nothing.
I tried d1_mini32, d32_pro, nothing.
  I haven't been able to do anything for 4 days.
arduino libraries tease me over it I couldn't flash, but sketch data upload works.

I would most like the java flash tool to work ..

do i have to buy d1 mini 32 or d32 pro board to work ...

wled works, I tried.

**Comment by MartinMueller2003** [comment]
Most of your links did not work. While you listed your CPU, you did not list the platform you are using. That will be important to the selection for which code to upload. I have attached the latest flashtool to this issue. It includes a save log button. 

1. Unzip it
2. double click the jar file
3. Select the Serial Port
4. select a hardware type
5. click upload
6. wait for it to finish and for the device to complete booting in the bottom pane
7. press save log, give it a name
8. post the log in this issue

[dist.zip](https://github.com/forkineye/ESPixelStick/files/8304396/dist.zip)
.

**Comment by ddijanic** [comment]
sorry for the links i corrected that now. uses only a boardless processor. I connected 3.3v en, gnd, rx, tx, u io0 to download mode.

because it's a problem, to take a regular d1 mini32?

**Comment by MartinMueller2003** [comment]
The problem is PSRAM. The D32 Pro uses PSRAM. I assume that you did not add PSRAM so you need to use one of the non PSRAM builds or create a custom build. d1_mini_mhetesp32minikit does not have PSRAM.

**Comment by MartinMueller2003** [comment]
The one FS that you managed to upload was in the wrong location and of the wrong type. It does not look like you are using the latest platformio file

**Comment by MartinMueller2003** [comment]
And Arduino cannot be used to generate the ESP32 images due to the use of a custom partition setup.

**Comment by ddijanic** [comment]
thanks for help.
here is the log file.
https://anotepad.com/notes/6pbq7mwy

I'm not an expert, as I see it after "Hard resetting via RTS pin ..."
no more connection to upload.
i think it's because there are no platforms with usb. rather it goes through programmers and RX and TX pins.
reset does not work properly, I tried to keep Io 0 on GND, not helping.
I need to get d1mini 32. if I'm not wrong.
I'll buy it on Tuesday.

**Comment by MartinMueller2003** [comment]
When you see this line "Connecting........_____....._____....._____....._____....._____....._____....._____", that indicates the program is waiting for you to reset the device back into the programming mode. That is where your sequence failed. The system needs you to enter the programming mode three times so you will need to babysit the sequence.

**Comment by ddijanic** [comment]
programming succeeded, with a manual reset between filesystem image and Firmware ..
the d1_mini_mhetesp32minikit works as you said.

a couple of questions.
Device setup page,
Primary Input: **artnet**, Output 1: no serial or DMX? (ws2811 / usc1903 / gs8208 only) I need **dmx**.
and Secondary Input: I need **artnet** again - Universe 2 and Output 2: DMX again ...
 
Supported-Inputs-and-Outputs says:
ESP32 platforms:

2 Series
6 Pixel outputs
8 Relay outputs
16 PWM Outputs (I2C bus)
2 dxm out and 2 pixel out, it would be perfect: D
but i don't see it in the Device setup page ...

and the last question
PSRAM is for better performance?
ako je tako , uzecu takav esp32...

**Comment by MartinMueller2003** [comment]
It is there. Not sure why you dont have it.

![image](https://user-images.githubusercontent.com/8608503/160903747-2627b058-9714-43c3-9e80-f770eab8fe3a.png)

**Comment by MartinMueller2003** [comment]
Input selection is independent from output selection. Notice I have no input selected at all and the complete output list is present.

**Comment by MartinMueller2003** [comment]
PSRAM allows us to have larger buffers and support more channels.

**Comment by ddijanic** [comment]
Okey. Thanks 
D1 mini32 Is okey ?
Psram, dual dmx out and dual or 4 pix out ?

**Comment by MartinMueller2003** [comment]
Depends on the Platform. There are a lot of D1 Mini variants but yes, that is the basics of it.

**Comment by ddijanic** [comment]
D1 mini ESP32 ESP-32 WiFi+Bluetooth Internet of Things development
https://a.aliexpress.com/_mtEWMZs
This Is ok ?
2x dmx and  2or4 pix out ?

**Comment by MartinMueller2003** [comment]
That is actually the MH ET Live board. It can support those outputs but you may need to adjust the GPIO settings.

**Comment by MartinMueller2003** [comment]
You should be fine using the D1 Mini ESP32 platform type.

**Comment by ddijanic** [comment]
Eps32 cam have PSRAM.
ITS Nice,i like it.
But no pin out for U2TXD 👎 

I need rx0 ,tx0, and rx1, tx1 for dual dmx out ..
______________________________________
ESP32 DEV KIT DevKitC v4 donest have PSRAM

or just tell me with ESP32 platform

**Comment by MartinMueller2003** [comment]
The ESP32-CAM board is the most unstable and IO constrained of all of the boards we support. I suggest avoiding it.

**Comment by MartinMueller2003** [comment]
I Have one and if I did not need it to verify the CAM builds it would be in the trash.

**Comment by ddijanic** [comment]
I'm so sorry to bother you @MartinMueller2003 , but I have a problem again.

I bought
https://www.aliexpress.com/item/4000090521976.html?spm=a2g0o.order_list.0.0.155d1802qgsoJj

the d1_mini_mhetesp32minikit is the only one that works, the others have psram error. of course: D: D

the d1_mini_mhetesp32minikit works, but I can't save anything on device output ..
here is the log list ...

[ESP] ESPixelStick v4.x-dev (Mar 17 2022 - 09:56:19)
[ESP] v4.4-beta1-308-gf3e0c8bc41
[FileMgr] Flash file system initialized. Used = 204800 out of 327680
[FileMgr] Configuration File: '/config.json' loaded.
[FileMgr] Configuration File: '/output_config.json' loaded.
[OutputMgr] Invalid Channel Type in config '7'. Specified for channel '0'. Disabling channel
[OutputMgr] Invalid Channel Type in config '7'. Specified for channel '1'. Disabling channel
[FileMgr] Configuration File: '/input_config.json' loaded.
[InputMgr] Invalid Channel Type in config '7'. Specified for channel '0'. Disabling channel
[InputMgr] Invalid Channel Type in config '7'. Specified for channel '1'. Disabling channel
[FileMgr] Configuration File: '/config.json' saved 522 bytes.
[WiFiDrv] WiFi Entering State: Connecting Using Config Credentials
[WiFiDrv] Using DHCP
[WiFiDrv] Connecting to 'Dijanic stan' as esps-28CF2655B594
[WiFiDrv] WiFi Entering State: Connected To AP
[WiFiDrv] Connected with IP: 192.168.1.5
[WebMgr] Web server listening on port 80
[FPPD] Listening on port 32320
[EthDrv] Entering State: Powering Up
[EthDrv] Entering State: Connecting
E (13422) esp.emac: emac_esp32_init (355): reset timeout
E (13423) esp_eth: esp_eth_driver_install (222): init mac failed
[13430] [E] [ETH_m.cpp: 310] begin (): esp_eth_driver_install failed
[EthDrv] Entering State: Device Init Failed
[28384] [E] [vfs_api.cpp: 102] open (): /littlefs/www/artnet.html does not exist, no permits for creation
[WebMgr] WS client connect - 1
[WebMgr] WS client disconnect - 1
[72572] [E] [vfs_api.cpp: 102] open (): /littlefs/www/favicon.ico.gz does not exist, no permits for creation
[WebMgr] WS client connect - 2
[WebMgr] WS client disconnect - 2
[94445] [E] [vfs_api.cpp: 102] open (): /littlefs/www/favicon.ico.gz does not exist, no permits for creation
[WebMgr] WS client connect - 3

and DXM out can only on one output, not me on two, output 1 and output 2 would like or 3.
just offers me on OUTPUT1 DMX, Serial, Renard

my uload log :

https://anotepad.com/notes/bqe5e5n4


https://prnt.sc/qgOqmvCla9Sd
https://prnt.sc/Q-iqxL3_6m30

**Comment by ddijanic** [comment]
you can just tell me which record to buy from aliexpress ..
I found this now this has a psram https://www.waveshare.com/nodemcu-esp-s3-12k-kit.htm
or esp32 ttgo t8?
i also need to run two dmx outputs.

**Reply by MartinMueller2003** [reply]
DMX is now supported out of all outputs (10 max). I still suggest using a lolin ESP 32 pro board. However, I have been integrating the QuinLed boards into V4 and the ESP32 AE version with the SD card hat is rock solid. You can make a ttl to differential converter (or use one of the pixel extenders that uses differential signaling) to get the DMX signaling and you are good to go.

**Comment by ddijanic** [comment]
Hay. Ttgo ti workd.
Now output 1,2,3,4 are DMX.
where can i see which are the pins?
and whether it can be set out 2 to be universe 2.
and out 3 unuvers 3.
we really need to adjust which university each output is...

![Screenshot_20220620_212321_com android chrome](https://user-images.githubusercontent.com/74422218/174667214-120a51d5-eb8f-4cce-9e71-ea27fc44cc68.jpg)

this seems to me that every dmx out is 1 universe.

**Reply by MartinMueller2003** [reply]
On the admin page, click advanced. Go back to config page and you will see the current pin definitions.

**Reply by MartinMueller2003** [reply]
As for assigning a universe to a port: My preference on that is NO NO NO. E1.31 and Artnet with the stupid universe definitions should be shot. With that said, if Shelby tells me I have to do something so silly then I will. Otherwise NO.

**Reply by ddijanic** [reply]
The all outputs are universe 1 automatically ?

**Reply by MartinMueller2003** [reply]
The default starting universe is 1 but you can change that to any universe. V4 processes the needed channels starting at the E1.31 universe/channel you define. If you are using 10 channels per port and 5 ports then it will process 50 channels from the E1.31 data stream. If you set the starting universe to 3, number of channels in a universe to 512 and the first channel offset to 500 then V4 will take data from universe three, channels 500 through 512 (thirteen channels) and universe four, channels 0 through 36 for a total of 50 channels.
